pkg/account/deployer/client.go

```diff
@@ -145,54 +146,89 @@ func (d *accountManagementClient) upsertPolicy(ctx context.Context, policyLevel
-		return "", err
+	if err = d.handleClientResponseError(resp, err, "unable to get group with ID: "+groupID); err != nil {
+		return nil, err
 	}
 
-	// find groups with matching name
-	var existingGroups []accountmanagement.GetGroupDto
 	for _, g := range result.GetItems() {
-		if g.GetName() == group.Name {
-			existingGroups = append(existingGroups, g)
+		if g.GetUuid() == groupID {
+			return &g, nil
 		}
 	}
 
-	if len(existingGroups) == 0 {
-		var createdGroups []accountmanagement.GetGroupDto
-		createdGroups, resp, err = d.client.GroupManagementAPI.CreateGroups(ctx, d.accountInfo.AccountUUID).PutGroupDto([]accountmanagement.PutGroupDto{group}).Execute()
-		defer closeResponseBody(resp)
-		if err = d.handleClientResponseError(resp, err, "unable to create group with name: "+group.Name); err != nil {
-			return "", err
-		}
-		if len(createdGroups) < 1 {
-			return "", fmt.Errorf("unable to get UUID of created group with name: %s", group.Name)
+	return nil, fmt.Errorf("unable to get group with ID: %s", groupID)
+}
+
+func (d *accountManagementClient) getGroupsByName(ctx context.Context, name string) ([]accountmanagement.GetGroupDto, error) {
+	groupList, resp, err := d.client.GroupManagementAPI.GetGroups(ctx, d.accountInfo.AccountUUID).Execute()
+	defer closeResponseBody(resp)
+	if err = d.handleClientResponseError(resp, err, "unable to get group with name: "+name); err != nil {
+		return nil, err
+	}
+
+	var groupsMatchingName []accountmanagement.GetGroupDto
+	for _, g := range groupList.GetItems() {
+		if g.GetName() == name {
+			groupsMatchingName = append(groupsMatchingName, g)
 		}
-		return createdGroups[0].GetUuid(), nil
 	}
 
-	if len(existingGroups) > 1 { // shouldn't happen
-		logr.FromContextOrDiscard(ctx).V(-1).Info("Updating multiple policies with name " + group.Name + ". Updating group with UUID " + existingGroups[0].GetUuid())
+	return groupsMatchingName, nil
+}
+
+func (d *accountManagementClient) createGroup(ctx context.Context, group Group) (remoteId, error) {
+	var createdGroups []accountmanagement.GetGroupDto
+	createdGroups, resp, err := d.client.GroupManagementAPI.CreateGroups(ctx, d.accountInfo.AccountUUID).PutGroupDto([]accountmanagement.PutGroupDto{group}).Execute()
+	defer closeResponseBody(resp)
+	if err = d.handleClientResponseError(resp, err, "unable to create group with name: "+group.Name); err != nil {
+		return "", err
 	}
+	if len(createdGroups) < 1 {
+		return "", fmt.Errorf("unable to get UUID of created group with name: %s", group.Name)
+	}
+	return createdGroups[0].GetUuid(), nil
+}
 
-	groupToUpdate := existingGroups[0]
+func (d *accountManagementClient) updateExistingGroup(ctx context.Context, existingGroup accountmanagement.GetGroupDto, group Group) (remoteId, error) {
+	// Groups with owner "SCIM" or "ALL_USERS" cannot be modified and so updates should be skipped
+	if featureflags.SkipReadOnlyAccountGroupUpdates().Enabled() && ((existingGroup.Owner == "SCIM") || (existingGroup.Owner == "ALL_USERS")) {
```

Comment:
could we logically turn this around and only update `LOCAL` groups (may be mis-remebering the exact owner for groups from DT's permission management)?

Reply:
Good point, however I'm not sure whether restrictions also apply to groups owned by `SAML` or `DCS`, and so restricted this to not update those I was told cannot be updated.

Reply:
Makes sense - I wasn't aware there's that many different types of owners.